nuitka/nodes/ImportNodes.py

```diff
@@ -39,6 +39,7 @@
 from nuitka.importing.ImportResolving import resolveModuleName
 from nuitka.importing.Recursion import decideRecursion
 from nuitka.importing.StandardLibrary import isStandardLibraryPath
+from nuitka.ModuleRegistry import addMissingModule
```

Comment:
That approach will never work well with caching, missing modules must be attached to a module. We will want to know what module, and where (source_ref), missed a module. In the case of caching we need to restore that information.